RHIDP-1454: Document TechDocs plugin (#294)
redhat-developer/red-hat-developers-documentation-rhdh: modules/getting-started/proc-techdocs-configure-odf-helm.adoc

[id="proc-techdocs-configure-odf-helm_{context}"]

> = Adding access information to the {product} Helm chart

Reviewer: I know this is the headline I proposed, but I still think it is a bit weird, and I don't know how to improve it :-(. In this section, we are configuring Helm Chart in a way that makes all information (that is stored in Secret and ConfigMap) about ObjectBucket available in the container as environment variables. Those variables are then used in TechDocs plugin configuration

> Add access information for the `ObjectBucket` to the {product} Helm chart configuration.

Reviewer: We should clarify that this is only required if Helm Chart was used to deploy RHDH. And if they are using RHDH Operator than they should follow the next section and skip this one.

> [id="proc-techdocs-configure-odf-helm_{context}"]
> 
> = Making object storage accessible to containers using the Helm chart

Reviewer 2: ```suggestion
// Module included in the following assemblies:
//
// * assemblies/assembly-techdocs-plugin.adoc

:_mod-docs-content-type: PROCEDURE
[id="proc-techdocs-configure-odf-helm_{context}"]
= Making object storage accessible to containers using the Helm chart
```

`ObjectBucket` information is stored in the {product-short} `ObjectBucketClaim` config map and secret. Adding the `ObjectBucket` information to the Helm chart configuration makes it accessible to the {product-short} container by adding the following environment variables to the container:

* `BUCKET_NAME`
* `BUCKET_HOST`
* `BUCKET_PORT`
* `BUCKET_REGION`
* `BUCKET_SUBREGION`
* `AWS_ACCESS_KEY_ID`
* `AWS_SECRET_ACCESS_KEY`

These variables are then used in the TechDocs plugin configuration.

.Prerequisites

* You have installed {product} on {ocp-short} using the Helm chart.

> * You have created an `ObjectBucketClaim` CR for storing files generated by TechDocs.

Reviewer 2: ```suggestion
* You have created an `ObjectBucketClaim` custom resource (CR) for storing files generated by TechDocs.
```
I'd spell out what CR is for the first use in a module, however, this reads a little confusing to me considering the intro line - which CR are we referring to, the config map, or the secret, both, or a different CR entirely? Do we have docs maybe we can link to about creating this CR that would make this clearer?

.Procedure

> * In the `upstream.backstage` key in the Helm chart values, enter the name of the Developer Hub `ObjectBucketClaim` secret as the value for the `extraEnvVarsSecrets` field and the `extraEnvVarsCM` field. For example:

Reviewer 2: ```suggestion
* In the `upstream.backstage` key in the Helm chart values, enter the name of the {product-short} `ObjectBucketClaim` secret as the value for the `extraEnvVarsSecrets` field and the `extraEnvVarsCM` field. For example:
```

+

> [source]

Reviewer 2: ```suggestion
[source,yaml]
```
The reason for using `[source]` on the codeblocks is so that we can use the syntax highlighting by specifying which language the code block is in. If we're not specifying the languages, you don't actually need to include these, but I think it's nice to have the right language added.
